src/mlpack/methods/bayesian_ridge/bayesian_ridge.cpp

```diff
@@ -0,0 +1,299 @@
+  fitIntercept(fitIntercept),
+  normalize(normalize)
+{/* Nothing to do */}
+
+float BayesianRidge::Train(const arma::mat& data,
+                           const arma::rowvec& responses)
+{
+  Timer::Start("bayesian_ridge_regression");
+
+  arma::mat phi;
+  arma::rowvec t;
+  arma::colvec vecphitT;
+  arma::mat phiphiT;
+  arma::colvec eigval;
+  arma::mat eigvec;
+  arma::colvec eigvali;
+
+  // Preprocess the data. Center and normalize.
+  CenterNormalize(data,
+                  responses,
+                  fitIntercept,
+                  normalize,
+                  phi,
+                  t,
+                  data_offset,
+                  data_scale,
+                  responses_offset);
+  vecphitT = phi * t.t();
+  phiphiT =  phi * phi.t();
+
+  arma::eig_sym(eigval, eigvec, phiphiT);
+
+  // Detect singular matrix.
+  if (eigval[0] == 0)
+  {
+    Log::Warn << "Singular matrix. Two lines or more are colinear."
+              <<  std::endl;
+    return -1;
+  }
+
+  unsigned short p = data.n_rows, n = data.n_cols;
+  // Initialize the hyperparameters and
+  // begin with an infinitely broad prior.
+  alpha = 1e-6;
+  beta =  1 / (var(t) * 0.1);
+
+  double tol = 1e-3;
+  unsigned short nIterMax = 50;
+  unsigned short i = 0;
+  double deltaAlpha = 1, deltaBeta = 1, crit = 1;
+  arma::mat matA = arma::eye<arma::mat>(p, p);
+  arma::rowvec temp;
+
+  while ((crit > tol) && (i < nIterMax))
+    {
+      deltaAlpha = -alpha;
+      deltaBeta = -beta;
+
+      // Compute the posterior statistics.
+      // with inv()
+      for (size_t k = 0; k < p; k++) {matA(k, k) = alpha;}
+      // inv is used instead of solve beacause we need matCovariance to
+      // compute the prediction uncertainties. If solve is used, matCovariance
+      // must be comptuted at the end of the loop.
+      matCovariance = inv_sympd(matA + phiphiT * beta);
+      omega = (matCovariance * vecphitT) * beta;
+
+      // // with solve()
+      // for (size_t k = 0; k < p; k++) {matA(k,k) = alpha / beta;}
+      // omega = solve(matA + phiphiT, vecphitT);
+
+      // Update alpha.
+      eigvali = eigval * beta;
+      gamma = sum(eigvali / (alpha + eigvali));
+      alpha = gamma / dot(omega.t(), omega);
+
+      // Update beta.
+      temp = t - omega.t() * phi;
+      beta = (n - gamma) / dot(temp, temp);
+
+      // Comptute the stopping criterion.
+      deltaAlpha += alpha;
+      deltaBeta += beta;
+      crit = abs(deltaAlpha/alpha + deltaBeta/beta);
+      i++;
+    }
+  Timer::Stop("bayesian_ridge_regression");
+  return Rmse(data, responses);
+}
+
+void BayesianRidge::Predict(const arma::mat& points,
+                            arma::rowvec& predictions) const
+{
+  arma::mat X = points;
+
+  // Center and normalize the points before applying the model
+  X.each_col() -= data_offset;
+  X.each_col() /= data_scale;
+  predictions = omega.t() * X + responses_offset;
+}
+
+void BayesianRidge::Predict(const arma::colvec& point,
```

Comment:
Usually we don't provide an interface for a single point, I think a user could just use a vector of size on and access the first index to achieve the same. Let me know what you think.

Reply:
I agree with you.I will delete it.